torchtext/experimental/vocab.py

```diff
@@ -0,0 +1,145 @@
+                                     "Please ensure that the `ordered_dict` doesn't contain any special tokens.".format(token))
+                tokens.append(token)
+
+        # assume special tokens dont appear in ordered_dict
+        if specials_first:
+            tokens = list(specials) + tokens
+        else:
+            tokens += list(specials)
+
+        self.vocab = torch.classes.torchtext.Vocab(tokens, unk_token)
+
+    @torch.jit.export
+    def __len__(self) -> int:
+        r"""Returns:
+            length (int): the length of the vocab
+        """
+        return len(self.vocab)
+
+    @torch.jit.export
+    def __getitem__(self, token: str) -> int:
+        r"""
+        Args:
+            token (str): the token used to lookup the corresponding index.
+
+        Returns:
+            index (int): the index corresponding to the associated token.
+        """
+        return self.vocab[token]
+
+    @torch.jit.export
+    def insert_token(self, token: str, index: int) -> None:
+        r"""
+        Args:
+            token (str): the token used to lookup the corresponding index.
+            index (int): the index corresponding to the associated token.
+
+        Raises:
+            RuntimeError: if `index` not between [0, Vocab.size()] or if token already exists in the vocab.
+        """
+        self.vocab.insert_token(token, index)
+
+    @torch.jit.export
+    def append_token(self, token: str) -> None:
+        r"""
+        Args:
+            token (str): the token used to lookup the corresponding index.
+        """
+        self.vocab.append_token(token)
+
+    @torch.jit.export
+    def lookup_token(self, index: int) -> str:
+        r"""
+        Args:
+            index (int): the index corresponding to the associated token.
+
+        Returns:
+            token (str): the token used to lookup the corresponding index.
+
+        Raises:
+            RuntimeError: if `index` not between [0, itos.size()].
+        """
+        return self.vocab.lookup_token(index)
+
+    @torch.jit.export
+    def lookup_tokens(self, indices: List[int]) -> List[str]:
+        r"""
+        Args:
+            indices (List[int]): the `indices` used to lookup their corresponding`tokens`.
+
+        Returns:
+            tokens (List[str]): the `tokens` associated with `indices`.
+
+        Raises:
+            RuntimeError: if an index within `indices` is not between [0, itos.size()].
+        """
+        return self.vocab.lookup_tokens(indices)
+
+    @torch.jit.export
+    def lookup_indices(self, tokens: List[str]) -> List[int]:
+        r"""
+        Args:
+            tokens (List[str]): the tokens used to lookup their corresponding `indices`.
+
+        Returns:
+            indices (List[int]): the 'indices` associated with `tokens`.
+        """
+        return self.vocab.lookup_indices(tokens)
+
+    @torch.jit.export
+    def get_stoi(self) -> Dict[str, int]:
```

Comment:
Do we really still want these?

Reply:
I think it made testing the functions a lot easier. Now I'm not sure how often the list of tokens inside the vocab will be needed by the end user. Do you see any negatives in keeping this?